.github/workflows/tests.yml

```diff
@@ -62,13 +61,20 @@ jobs:
         python-version: ${{ matrix.python }}
         cache: pip
 
-    - name: Install and upgrade tox
+    - name: Install and upgrade test tools
       run: python -m pip install --progress-bar off --upgrade tox
 
+    - name: Cache
+      uses: actions/cache@v4
+      with:
+        path: |
+          .tox
+        key: ${{ matrix.toxenv }}-${{ runner.os }}-${{ hashFiles('requirements.txt', 'pyproject.toml') }}
+
     - name: Run tests
-      run: tox ${{ matrix.toxargs }} -e ${{ matrix.toxenv }} -- ${{ matrix.toxposargs }}
```

Comment:
We weren't using `toxargs` or `toxposargs` except for the `--develop` flag, but that's been removed.  We can instead define flags to pass to pytest, etc. in the `tox` environments themselves.